src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -185,6 +189,31 @@ private void updateResources()
         final StructureName sn = new StructureName(Structures.SCHEMATICS_PREFIX, styles.get(stylesDropDownList.getSelectedIndex()) ,
                 building.getSchematicName() + nextLevel);
         final StructureWrapper wrapper = new StructureWrapper(world, sn.toString());
+        final Structure structure = wrapper.getStructure().getStructure();
+        final String md5 = Structures.getMD5(sn.toString());
+        if (structure.isTemplateMissing() || !structure.isCorrectMD5(md5))
+        {
+            if (structure.isTemplateMissing())
+            {
+                Log.getLogger().info("Template structure " + sn + " missing");
+            }
+            else
+            {
+                Log.getLogger().info("structure " + sn + " md5 error");
+            }
+
+            Log.getLogger().info("Request To Server for structure " + sn);
+            if (FMLCommonHandler.instance().getMinecraftServerInstance() == null)
+            {
+                MineColonies.getNetwork().sendToServer(new SchematicRequestMessage(sn.toString()));
+                return;
+            }
+            else
+            {
+                Log.getLogger().error("WindowBuildTool: Need to download schematic on a standalone client/server. This should never happen");
+            }
+        }
+
         wrapper.setPosition(building.getLocation());
         wrapper.rotate(building.getRotation(), world, building.getLocation(), building.isMirrored() ? Mirror.FRONT_BACK : Mirror.NONE);
         while (wrapper.findNextBlock())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)